### DISCUSSION TITLE
Are CConfigurationData and CConfigurationDataProvider currently the preferred/adviced way to do a MBS type builder integration into CDT?

### DISCUSSION THREAD

**Discussion by jantje** [eclipse-cdt/cdt#337] [Q&A]
Hi
I'm currently rewriting managed build. I have succeeded to get the project creation and build to work.
I wanted to add some GU things for testing; but I've been trying for weeks to get the UI to work 😒 . I run into problems deep inside CDT that I still fail to understand (F.I. in PathSettingsContainer) and those issues are not MBS specific.
However looking at the type hierarchies of these 2 classes I do not see other build providers implementing these classes.
Also following classes used in this solution have documentation issues

- CConfigurationData
- CFolderData
- CFileData
- CResourceData
- CDataObject
- ICSourceEntry
- ICExclusionPatternPathEntry
- ICPathEntry
- ICSettingEntry
- CBuildData
- CConfigurationStatus
- CTargetPlatformData

Before I try to explain the problems I'm running into (which would clearly demonstrate I have no clue what I'm doing 🤡 ) I would like to be sure this is the way to do a MBS type builder integration into CDT.
If not please point me to the appropriate way/documentation.
Best regards
Jantje

**Comment by jonahgraham** [comment]
Sorry that you haven't gotten any answers on this. I don't know if there is anyone who wrote that code (a long time ago) active enough to understand/explain it better than the source code itself.

**Reply by jantje** [reply]
I understand. But still the answer to the question "Are CConfigurationData and CConfigurationDataProvider currently the preferred/adviced way to do a MBS type builder integration into CDT?" should be answerable.
I mean, are there better/more adviced alternatives?

**Reply by jonahgraham** [reply]
It is answerable - but I don't know off the top of my head. I would have to examine the history of the code to answer it.

If any of the other @eclipse-cdt/tools-cdt-committers have ideas feel free to expand here.
